src/main/java/com/cloudbees/jenkins/support/impl/CustomLogs.java

```diff
@@ -0,0 +1,217 @@
+import java.util.logging.Logger;
+import java.util.logging.StreamHandler;
+
+/**
+ * Custom Log recorders files from the master.
+ */
+@Extension(ordinal = 100.0)
+public class CustomLogs extends Component {
+
+    private static final int MAX_ROTATE_LOGS = Integer.getInteger(CustomLogs.class.getName() + ".MAX_ROTATE_LOGS", 9);
+    private final Map<String,LogRecorder> logRecorders = Jenkins.get().getLog().logRecorders;
+    private final File customLogs = new File(getLogsRoot(), "custom");
+
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Jenkins.ADMINISTER);
+    }
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Master Custom Log Recorders";
+    }
+
+    @Override
+    public void start(@NonNull SupportContext context) {
+        Logger.getLogger("").addHandler(new CustomHandler());
+    }
+
+    @Override
+    public void addContents(@NonNull Container result) {
+        addLogRecorders(result);
+    }
+
+    /**
+     * Dumps the content of {@link LogRecorder}, which is the groups of loggers configured
+     * by the user. The contents are also ring buffer and only remembers recent 256 or so entries.
+     */
+    private void addLogRecorders(Container result) {
+        for (Map.Entry<String, LogRecorder> entry : logRecorders.entrySet()) {
+            String name = entry.getKey();
+            String entryName = "nodes/master/logs/custom/{0}.log"; // name to be filtered in the bundle
+            File storedFile = new File(customLogs, name + ".log");
+            if (storedFile.isFile()) {
+                result.add(new FileContent(entryName, new String[]{name}, storedFile));
+            } else {
+                // Was not stored for some reason; fine, just load the memory buffer.
+                final LogRecorder recorder = entry.getValue();
+                result.add(new LogRecordContent(entryName, new String[]{name}) {
+                    @Override
+                    public Iterable<LogRecord> getLogRecords() {
+                        return recorder.getLogRecords();
+                    }
+                });
+            }
+        }
+    }
+
+    /**
+     * Returns the root directory for logs (historically always found under <code>$JENKINS_HOME/logs</code>.
+     * Configurable since Jenkins 2.114.
+     *
+     * @see hudson.triggers.SafeTimerTask#LOGS_ROOT_PATH_PROPERTY
+     * @return the root directory for logs.
+     */
+    private File getLogsRoot() {
+        final String overriddenLogsRoot = System.getProperty("hudson.triggers.SafeTimerTask.logsTargetDir");
+        if (overriddenLogsRoot == null) {
+            return new File(Jenkins.get().getRootDir(), "logs");
+        } else {
+            return new File(overriddenLogsRoot);
+        }
+    }
+
+    @SuppressFBWarnings(value="SIC_INNER_SHOULD_BE_STATIC_NEEDS_THIS", justification="customLogs is not static, so this is a bug in FB")
```

Comment:
Actually it since valid. customLogs looks like a constant, so it should probably be static.